charmhelpers/contrib/charmsupport/nrpe.py

```diff
@@ -416,6 +417,15 @@ def add_init_service_checks(nrpe, services, unit_name, immediate_check=True):
     :param str unit_name: Unit name to use in check description
     :param bool immediate_check: For sysv init, run the service check immediately
     """
+    # check_haproxy is redundant in the presence of check_crm. See LP Bug#1880601 for details.
+    # just remove check_haproxy if haproxy is added as a lsb resource in hacluster.
+    for rid in relation_ids("ha"):
+        ha_resources = relation_get("json_resources", rid=rid, unit=local_unit())
+        if ha_resources:
+            if "lsb:haproxy" in json.loads(ha_resources).values():
```

Comment:
`json.loads(...)` may raise an error; would be good to put this in a `try: except:` block to catch the error and do a nice log message if there is a problem.  e.g. should the hook crash if this fails?

Reply:
> should the hook crash if this fails?

I think it should crash because something is very wrong if the ha relation doesn't set a string that can be converted into a dict.
 
I've added the try except block and added another unit test to raise the error.